drivers/mipi_dbi/mipi_dbi_nxp_flexio_lcdif.c

```diff
@@ -0,0 +1,485 @@
+
+	/* Now the data are in shifter, wait for the data send out from the shifter. */
+	FLEXIO_MCULCD_WaitTransmitComplete();
+
+	/* Disable the TX shifter and the timer. */
+	FLEXIO_MCULCD_ClearMultiBeatsWriteConfig(flexio_lcd);
+
+	/* De-assert nCS. */
+	FLEXIO_MCULCD_StopTransfer(flexio_lcd);
+
+	k_sem_give(&lcdif_data->transfer_done);
+}
+
+
+static void flexio_lcdif_set_cs(bool set, void *param)
+{
+	const struct device *flexio_dev = (struct device *)param;
+	const struct mcux_flexio_lcdif_config *config = flexio_dev->config;
+
+	gpio_pin_set_dt(&config->cs_gpio, (int)set);
+}
+
+static void flexio_lcdif_set_rs(bool set, void *param)
+{
+	const struct device *flexio_dev = (struct device *)param;
+	const struct mcux_flexio_lcdif_config *config = flexio_dev->config;
+
+	gpio_pin_set_dt(&config->rs_gpio, (int)set);
+}
+
+static void flexio_lcdif_set_rd_wr(bool set, void *param)
+{
+	const struct device *flexio_dev = (struct device *)param;
+	const struct mcux_flexio_lcdif_config *config = flexio_dev->config;
+
+	gpio_pin_set_dt(&config->rdwr_gpio, (int)set);
+}
+
+static edma_modulo_t flexio_lcdif_get_edma_modulo(uint8_t shifterNum)
+{
+	edma_modulo_t ret = kEDMA_ModuloDisable;
+
+	switch (shifterNum) {
+	case 1U:
+		ret = kEDMA_Modulo4bytes;
+		break;
+	case 2U:
+		ret = kEDMA_Modulo8bytes;
+		break;
+	case 4U:
+		ret = kEDMA_Modulo16bytes;
+		break;
+	case 8U:
+		ret = kEDMA_Modulo32bytes;
+		break;
+	default:
+		ret = kEDMA_ModuloDisable;
+		break;
+	}
+
+	return ret;
+}
+
+static void flexio_lcdif_write_data_array(FLEXIO_MCULCD_Type *base,
+					  const void *data,
+					  size_t size)
+{
+	assert(size > 0U);
+
+	uint32_t i;
+	const uint8_t *data8Bit;
+	FLEXIO_Type *flexioBase = base->flexioBase;
+
+	/* Assert the RS pin. */
+	base->setRSPin(true, base->userData);
+	/* For 6800, de-assert the RDWR pin. */
+	if (kFLEXIO_MCULCD_6800 == base->busType) {
+		base->setRDWRPin(false, base->userData);
+	}
+
+	/* Configure the timer and TX shifter. */
+	FLEXIO_MCULCD_SetSingleBeatWriteConfig(base);
+
+	data8Bit = (const uint8_t *)data;
+
+	for (i = 0; i < size; i++) {
+		flexioBase->SHIFTBUF[base->txShifterStartIndex] = data8Bit[i];
+
+		/* Wait for the data send out. */
+		while (0U == ((1UL << base->timerIndex) & flexioBase->TIMSTAT)) {
+		}
+
+		/* Clear the timer stat. */
+		flexioBase->TIMSTAT = 1UL << base->timerIndex;
+	}
+
+	/* Stop the timer and TX shifter. */
+	FLEXIO_MCULCD_ClearSingleBeatWriteConfig(base);
+}
+
+static int mipi_dbi_flexio_lcdif_configure(const struct device *dev,
+					   const struct mipi_dbi_config *dbi_config)
+{
+	const struct mcux_flexio_lcdif_config *config = dev->config;
+	struct mcux_flexio_lcdif_data *lcdif_data = dev->data;
+	flexio_mculcd_config_t flexioMcuLcdConfig;
+	int err;
+	uint32_t clock_freq;
+	uint32_t mipi_mode = dbi_config->mode;
+	status_t status;
+
+	/* 9-bit mode is not supported by the SDK driver */
+	if ((mipi_mode == MIPI_DBI_MODE_6800_BUS_9_BIT) ||
```

Comment:
We should probably add an additional check here to see if the requested MIPI mode is 8 bit when the driver is configured for 16 bit (or vice versa for 16 bit mode). Not *sure* if we need this though, since the CMake logic should identify the correct MIPI DBI setting based on the chosen display

Reply:
Thats right, the SDK driver configuration is picked from device tree.